### PR TITLE
Add `isDomainName`

### LINKED ISSUE #161: RFC: add assertion utility to test if a value is a domain name
## Checklist

> Please ensure the following tasks are completed before filing an issue.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
-   [x] If this is a feature request, the issue name begins with `RFC:`.

## Description

> Description of the issue (or feature request).

This RFC proposes adding an assertion utility which tests whether a value is a domain name. 

Prior art:

-   [is-domain-name](https://github.com/emilbayes/is-domain-name/blob/master/index.js): includes additional constraints which may not want to follow. Ideally, the utility should be RFC compliant.

Package: `@stdlib/assert/is-domain-name`
Alias: `isDomainName`

## Related Issues

> Does this issue (or feature request) have any related issues?

No.

## Questions

> Any questions for reviewers?

-   Are there other RFCs we need to consult beside RFC 2181?

## Other

> Any other information relevant to this issue (or feature request)? This may include screenshots, references, stack traces, sample output, and/or implementation notes.

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #161.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds an assertion utility for testing whether a value is a domain name

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #161 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/assert/is-domain-name/test/test.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		bool = isDomainName( values[i] );
+		t.equal( bool, false, 'returns false when provided '+values[i] );
+	}
+	t.end();
+});
+
+tape( 'the function returns `false` if not provided a string primitive', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-domain-name/test/test.js]:
```suggestion
tape( 'the function returns `false` if not provided a string', function test( t ) {
```

File: lib/node_modules/@stdlib/assert/is-domain-name/test/test.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+	t.end();
+});
+
+tape( 'the function returns `false` if provided an invalid domain name', function test( t ) {
+	var values;
+	var bool;
+	var i;
+
+	values = [
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-domain-name/test/test.js:65]: @Planeshifter To exercise the `2` and `255` character limits, should add tests which exceed these bounds. One for an empty string, another for a single char string, and another for a string longer than 255 chars.

**Review by kgryte — CHANGES_REQUESTED** [review]
LGTM apart from needing to test 3 edge cases to exercise all code paths.

**Comment by Planeshifter** [comment]
Thanks! Added test values.

File: lib/node_modules/@stdlib/assert/is-domain-name/lib/main.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+*
+* -   `[a-z0-9-]{0,61}`
+*     -   match between 0 and 61 digits, letters, or hyphens
+*
+* -   `[a-z0-9]`
+*     -   match a single digit or letter
+*
+* -   `$`
+*     -   match end of string
+*/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-domain-name/lib/main.js:73]:
```suggestion
*
* @private
* @constant
* @type {RegExp}
*/
```

As this is a JSDoc comment, we need to add the `private` tag.

File: lib/node_modules/@stdlib/assert/is-domain-name/package.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
+{
+  "name": "@stdlib/assert/is-domain-name",
+  "version": "0.0.0",
+  "description": "",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-domain-name/package.json]:
```suggestion
  "description": "Test if a value is a domain name.",
```

**Review by kgryte — APPROVED** [review]
LGTM

**Pull request merged**